Userland/Libraries/LibJS/Runtime/Temporal/ZonedDateTime.cpp

```diff
@@ -538,7 +538,23 @@ ThrowCompletionOr<NanosecondsToDaysResult> nanoseconds_to_days(VM& vm, Crypto::S
         }
     }
 
-    // 19. Return the Record { [[Days]]: days, [[Nanoseconds]]: nanoseconds, [[DayLength]]: abs(dayLengthNs) }.
+    // 19. If days < 0 and sign = 1, throw a RangeError exception.
+    if (days < 0 && sign == 1)
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidEpochNanoseconds);
+
+    // 20. If days > 0 and sign = -1, throw a RangeError exception.
+    if (days > 0 && sign == -1)
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidEpochNanoseconds);
+
+    // 21. If nanoseconds < 0 and sign = 1, throw a RangeError exception.
+    if (nanoseconds.is_negative() && sign == 1)
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidEpochNanoseconds);
+
+    // 22. If nanoseconds > 0 and sign = -1, throw a RangeError exception.
+    if (!nanoseconds.is_negative() && sign == -1)
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidEpochNanoseconds);
```

Comment:
This won't handle `nanoseconds = 0` correctly.

Also just for clarity you might want to add an explicit check for `0` in the if above (step 21)

Reply:
Sorry, I didn't get your second point.